server/sublist.go

```diff
@@ -548,7 +548,9 @@ func (s *Sublist) match(subject string, doLock bool, result *SublistResult) *Sub
 	if doLock {
 		s.RLock()
 	}
-	cacheEnabled := s.cache != nil
+	// Writing to the cache is only allowed if not supplying our
+	// own SublistResult, i.e. via call to MatchWithResult.
+	cacheEnabled := result == nil && s.cache != nil
```

Comment:
By the way, the other danger is that - although in this PR we make sure we don't store the result in the cache - we still may get a cache hit from a previously stored result (from some other code doing a regular match, with result being cached). If we get a hit, we will return the result from the cache, which is fine with the way you have use it in the other PR. However, if later some dev thinks that the returned result is the same than the pointer to the SublistResult from the stack, bad things will happen. Say:
```
r := &SublistResult{}
for _, subj := range []subjectsToCheck {
    if r = sl.MatchWithResult(subj, r); len(r.psubs) > 0 {
        // do something
    }
}
```
The code above would be bad if one result was returned from a cache hit, because at the next invocation in the loop, we would reset the cache result's content.

This is a far-fetched misuse, but we have to be mindful of the consequences.